src/main/java/io/anserini/collection/AclCollection.java

```diff
@@ -0,0 +1,201 @@
+        if (iter.hasNext()) {
+          nodeEntry = iter.next();
+        }
+      }
+    }
+
+    @Override
+    public void readNext() throws NoSuchElementException {
+      if (nodeEntry == null) {
+        throw new NoSuchElementException("JsonNode is empty");
+      } else {
+        bufferedRecord = new AclCollection.Document(nodeEntry);
+        if (iter.hasNext()) {
+          nodeEntry = iter.next();
+        } else {
+          atEOF = true; // there is no more JSON object in the bufferedReader
+        }
+      }
+    }
+  }
+
+  /**
+   * A document in a JSON collection.
+   */
+  public class Document implements SourceDocument {
+    private String id;
+    private String contents;
+    private JsonNode paper;
+    private JsonNode volume;
+    private List<String> authors;
+    private List<String> venues;
+    private List<String> sigs;
+
+    public Document(Map.Entry<String, JsonNode> jsonEntry) {
+      id = jsonEntry.getKey();
+      paper = jsonEntry.getValue();
+      contents = getOrDefault(paper, "title") + " " + getOrDefault(paper, "abstract_html");
+
+      // Process author facets
+      authors = new ArrayList<>();
+
+      if (paper.has("author")) {
+        ArrayNode authorNode = (ArrayNode) paper.get("author");
+        authorNode.elements().forEachRemaining(node -> {
+          authors.add(((ObjectNode) node).get("full").asText());
+        });
+      }
+
+      // Retrieve parent volume metadata
+      String parentVolumeId = paper.get("parent_volume_id").asText();
+      volume = AclCollection.this.volumes.get(parentVolumeId);
+
+      // Process venue facets
+      venues = new ArrayList<>();
+      ArrayNode venuesNode = (ArrayNode) volume.get("venues");
+      venuesNode.elements().forEachRemaining(node -> {
+        venues.add(node.asText());
+      });
+
+      // Process SIG facets
+      sigs = new ArrayList<>();
+      ArrayNode sigsNode = (ArrayNode) volume.get("sigs");
+      sigsNode.elements().forEachRemaining(node -> {
+        sigs.add(node.asText());
+      });
+    }
+
+    @Override
+    public String id() {
+      if (id == null) {
+        throw new RuntimeException("JSON document has no \"id\" field");
+      }
+      return id;
+    }
+
+    @Override
+    public String content() {
+      if (contents == null) {
+        throw new RuntimeException("JSON document has no \"contents\" field");
+      }
+      return contents;
+    }
+
+    @Override
+    public boolean indexable() {
+      return true;
+    }
+
+    public JsonNode paper() {
+      return paper;
+    }
+
+    public JsonNode volume() {
+      return volume;
+    }
+
+    public List<String> authors() {
+      return authors;
+    }
+
+    public List<String> venues() {
+      return venues;
+    }
+
+    public List<String> sigs() {
+      return sigs;
+    }
+
+    private String getOrDefault(JsonNode json, String key) {
+      if (json.has(key)) {
+        return json.get(key).asText();
+      }
+      return "";
```

Comment:
return `key`?

Reply:
Doesn't make sense because if the key,value pair is not the map, why would you return the key itself?